File: src/Medium/TODOLeetcode5.java
package Medium;

public class TODOLeetcode5 {
    public String longestPalindrome(String s) {
        int start=0;
        int end=0;
        for(int i=0;i<s.length();i++){
            int pali=Math.max(getPali(s,i,i),getPali(s,i,i+1));
            if(pali>end-start){
                start=i-(pali-1)/2;
                end=i+pali/2;
            }
        }
        return s.substring(start,end+1);
    }
    public int getPali(String s, int left, int right){
        while(left>=0 && right<s.length() && s.charAt(left)==s.charAt(right)){
            left--;
            right++;
        }
        return right-left-1;
    }
}

File: src/Python/Medium/TODOLeetcode5.py
#analyze the return statement in longestPalindrome method
class Solution:
    def pali(self,s,left,right):
        while(left>=0 and right<len(s) and s[left]==s[right]):
            left-=1
            right+=1
        return right-left-1

    def longestPalindrome(self, s: str) -> str:
        ansLen=0
        ansMid=0
        for i in range(len(s)):
            odd=self.pali(s,i,i)
            if odd>ansLen:
                ansLen=odd
                ansMid=i

            even=self.pali(s,i,i+1)
            if even>ansLen:
                ansLen=even
                ansMid=i

        return s[ansMid-(ansLen-1)//2:ansMid+(ansLen+2)//2]